### PR TITLE
link to nn libraries in readme

### PR DESCRIPTION
**Pull request by mattjj**
The goal here is to help users find the right tools for them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -460,6 +462,11 @@ if you run into any errors or problems with the prebuilt wheels.
 See [Building JAX from
 source](https://jax.readthedocs.io/en/latest/developer.html#building-from-source).
 
+## Neural network libraries
+
+There are lots of great deep learning libraries built on top of JAX!
+
+If you want a batteries-included library for neural network training, try [Flax](https://github.com/google/flax) by a Google Brain team in Amsterdam. Another good option is DeepMind's [Haiku](https://github.com/deepmind/dm-haiku), a JAX version of Sonnet focused solely on neural network layers. Finally, [Trax](https://github.com/google/trax) by a Brain team in California is a configuration-driven framework focused on sequence model research and reinforcement learning as a successor to Tensor2Tensor.
```
Comment (gnecula) [inline README.md:469]: Should we leave out the Amsterdam part? Anselm is a contributor and is not in Amsterdam. WDYT?

**Review by gnecula — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mattjj) [reply README.md:469]: True! Anselm is a bit of a globetrotter. Though still most developers are in AMS.

We'll bounce the wording off the Flax folks before finalizing anything. Same for the Haiku wording.

File: README.md [hunk 2]
```diff
@@ -460,6 +462,11 @@ if you run into any errors or problems with the prebuilt wheels.
 See [Building JAX from
 source](https://jax.readthedocs.io/en/latest/developer.html#building-from-source).
 
+## Neural network libraries
+
+There are lots of great deep learning libraries built on top of JAX!
```
Comment (gnecula) [inline README.md:467]: This sentence, and especially the `lots` part, can be interpreted in different ways by different people. I suggest skipping this intro and going directly to the descriptions.

File: README.md (hunk 1 above)
Comment (gnecula) [inline README.md:469]: I think that "batteries-included" would not be clear to some of the readers.

**Pull request merged**